server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -818,18 +823,36 @@ public void run() {
             updateCounts.put(ke, tablet.getUpdateCount());
           });
 
-          // gather metadata for all tablets readTablets()
-          try (TabletsMetadata tabletsMetadata =
-              getContext().getAmple().readTablets().forTablets(onlineTabletsSnapshot.keySet())
-                  .fetch(FILES, LOGS, ECOMP, PREV_ROW).build()) {
-
-            // for each tablet, compare its metadata to what is held in memory
-            tabletsMetadata.forEach(tabletMetadata -> {
-              KeyExtent extent = tabletMetadata.getExtent();
-              Tablet tablet = onlineTabletsSnapshot.get(extent);
-              Long counter = updateCounts.get(extent);
-              tablet.compareTabletInfo(counter, tabletMetadata);
-            });
+          Instant start = Instant.now();
+          Duration duration = Duration.ZERO;
+          int tabletCount = 0;
+          Span mdScanSpan = TraceUtil.startSpan(this.getClass(), "metadataScan");
+          try (Scope scope = mdScanSpan.makeCurrent()) {
+            // gather metadata for all tablets readTablets()
+            try (TabletsMetadata tabletsMetadata =
+                getContext().getAmple().readTablets().forTablets(onlineTabletsSnapshot.keySet())
+                    .fetch(FILES, LOGS, ECOMP, PREV_ROW).build()) {
+              mdScanSpan.end();
+              duration = Duration.between(start, Instant.now());
+
+              // for each tablet, compare its metadata to what is held in memory
+              for (var tabletMetadata : tabletsMetadata) {
+                tabletCount++;
+                KeyExtent extent = tabletMetadata.getExtent();
+                Tablet tablet = onlineTabletsSnapshot.get(extent);
+                Long counter = updateCounts.get(extent);
+                tablet.compareTabletInfo(counter, tabletMetadata);
+              }
+
+              log.debug("Metadata scan took {}ms for {} tablets read.", duration.toMillis(),
+                  tabletCount);
+            }
+          } finally {
+            if (duration.toMinutes() > 1) {
+              log.warn(
+                  "Metadata scan took {}ms for {} tablets. Performance of all activities will be severely hindered!",
+                  duration.toMillis(), tabletCount);
+            }
```

Comment:
I am not sure if we want to keep the warning. I don't know if it will ever get above 1 minute ore how useful it is once the performance gets that bad. At least with the debug statement we can get an idea of the performance on a large cluster.

Reply:
yeah maybe the warning could be dropped with the debug stmt.  Could move the debug stmt closer to the metadata read in case there is an exception.

Reply:
Doing some local testing with Uno, this is the best (or worst) I was able to do:
<pre>
tserver2_ip-10-113-12-25.log:2022-03-23T13:07:31,950 [tserver.TabletServer] DEBUG: Metadata scan took 1019ms for 42303 tablets read.
</pre>

Reply:
> Could move the debug stmt closer to the metadata read in case there is an exception.

Then I would have to remove the `tabletCount` from the debug message. I guess I could have 2 separate debug statements.